dunwoody_disaster/views/action_selector.py

```diff
@@ -2,29 +2,100 @@
 from PySide6.QtGui import QPixmap
 from typing import Optional
 from dunwoody_disaster import Item
+from dunwoody_disaster.CharacterFactory import Character
 import dunwoody_disaster as DD
+import random
 
 
 class ActionSelector(QWidget):
-    def __init__(self):
+    def __init__(self, character: Character):
         super().__init__()
+        self.character = character
         self.attack: Optional[Item.Weapon] = None
         self.defense: Optional[Item.Armor] = None
         self.setLayout(self.createLayout())
 
+    def clear(self):
+        self.setAttack(None)
+        self.setDefense(None)
+
     def setAttack(self, item: Optional[Item.Weapon]):
+        if item is not None:
+            staminaCost = item.staminaCost
+            magicCost = item.magicReq
+            # TODO: Implement
+            # if self.defense:
+            #     staminaCost += self.defense.staminaCost
+            #     magicCost += self.defense.magicReq
```

Comment:
Has this been implemented?

Reply:
No.
The reason I didn't implement this was because I don't think defenses have a stamina or magic cost.